propka/output.py

```diff
@@ -3,6 +3,42 @@
 from propka.lib import info
 
 
+def open_file_for_writing(input_file):
+    """Open file or file-like stream for writing.
+
+    TODO - convert this to a context manager.
+
+    Args:
+        input_file: path to file or file-like object. If file-like object,
+        then will attempt to get file mode.
+    """
+    try:
+        mode = input_file.mode
+        if not ("w" in mode or "a" in mode or "+" in mode):
```

Comment:
I think you could also use the [writable()](https://docs.python.org/3/library/io.html#io.IOBase.writable) method:
```python
        if input_file.writable():
```

Reply:
Sweet!  That's a new one for me.  I'll make the change.  Thanks.